content/pytorch/concepts/tensors/terms/rand/rand.md

```diff
@@ -0,0 +1,106 @@
+
+```
+
+`torch.rand()` take one parameter `size` which represent the shape of the tensor that wish to create. It can create a tensor with one, two, or more dimensions.
+
+Meanwhile, `torch.rand()` have other seven optional arguments that can be used to customize the tensor if needed.
+
+- `generator` is used for generating number for sampling.
+- `out`, to store the output into existing tensor (if there is any).
+- `dtype` is used to customize the data type of the output.
+- `layout`, to specify how Tensor is stored in memory.
+- `device`, to specify where the tensor will be created.
+- `requires_grad`, to record the operation of the tensor.
+- `pin_memory` is used for allocating the tensor in the pinned memory.
+
+## Example
+
+The example below demonstrates how the `size` parameter can be applied to create three different types of tensors. `tensor1D` is a one-dimensional tensor, or vector, with a size of 3; `tensor2D` is a two-dimensional tensor, typically called a matrix; and `tensor3D` is a three-dimensional tensor.
+
+```py
+import torch
+
+# Create a one-dimensional tensor of size three.
+tensor1D = torch.rand(3)
+print(tensor1D)
+
+# Create a two-dimensional tensor of size five by five.
+tensor2D = torch.rand(5, 5)
+print(tensor2D)
+
+# Create a three-dimensional tensor of three by five by five tensor
+tensor3D = torch.rand(3, 5, 5)
+print(tensor3D)
+
+```
+
+The outputs of `tensor1D`, `tensor2D`, and `tensor3D` are shown below, respectively.
+
+**Please note** that the result may not be exactly the same as the one displayed below, due to random number generation.
+
+`tensor1D`:
+
+```shell
+tensor([0.4581, 0.8923, 0.9110])
+```
+
+`tensor2D`:
+
+```shell
+tensor([[0.9166, 0.4592, 0.1805, 0.5239, 0.4623],
+        [0.8937, 0.9080, 0.9411, 0.5834, 0.8225],
+        [0.4368, 0.9289, 0.7732, 0.7876, 0.5329],
+        [0.2059, 0.1736, 0.4619, 0.4613, 0.2558],
+        [0.7025, 0.7887, 0.0750, 0.7822, 0.7054]])
+```
+
+`tensor3D`:
+
+```shell
+tensor([[[0.7478, 0.4375, 0.9769, 0.3897, 0.9407],
+         [0.9649, 0.6459, 0.7927, 0.1902, 0.9625],
+         [0.1942, 0.4772, 0.5795, 0.5631, 0.1460],
+         [0.9176, 0.5342, 0.1067, 0.1989, 0.9478],
+         [0.2999, 0.2278, 0.8297, 0.4244, 0.2445]],
+
+        [[0.0626, 0.1151, 0.2487, 0.0667, 0.9877],
+         [0.2304, 0.8230, 0.7293, 0.6631, 0.9981],
+         [0.3059, 0.0967, 0.4872, 0.8695, 0.2324],
+         [0.6734, 0.6645, 0.8859, 0.0637, 0.7380],
+         [0.3292, 0.3411, 0.0896, 0.2175, 0.4013]],
+
+        [[0.2818, 0.3043, 0.1920, 0.2259, 0.5281],
+         [0.2549, 0.5776, 0.6986, 0.3627, 0.7794],
+         [0.7877, 0.7343, 0.9829, 0.8692, 0.5039],
+         [0.7503, 0.5880, 0.6443, 0.8743, 0.0187],
+         [0.9316, 0.5449, 0.4706, 0.1604, 0.7445]]])
+```
+
+Though one- and two-dimensional tensors are well-known, there are instances in machine learning where a three-dimensional tensor is essential.
```

Comment:
Modifications:

- Let's merge the outputs into a single output
- Let's change the introduction for the output a bit
- Let's move the note from before the output to after the output
- Let's start the note with `> Note:` as notes should be started this way
- Let's change the note text and the last sentence a bit

````suggestion
Here is the output for the above example:

```shell
tensor([0.4581, 0.8923, 0.9110])

tensor([[0.9166, 0.4592, 0.1805, 0.5239, 0.4623],
        [0.8937, 0.9080, 0.9411, 0.5834, 0.8225],
        [0.4368, 0.9289, 0.7732, 0.7876, 0.5329],
        [0.2059, 0.1736, 0.4619, 0.4613, 0.2558],
        [0.7025, 0.7887, 0.0750, 0.7822, 0.7054]])

tensor([[[0.7478, 0.4375, 0.9769, 0.3897, 0.9407],
         [0.9649, 0.6459, 0.7927, 0.1902, 0.9625],
         [0.1942, 0.4772, 0.5795, 0.5631, 0.1460],
         [0.9176, 0.5342, 0.1067, 0.1989, 0.9478],
         [0.2999, 0.2278, 0.8297, 0.4244, 0.2445]],

        [[0.0626, 0.1151, 0.2487, 0.0667, 0.9877],
         [0.2304, 0.8230, 0.7293, 0.6631, 0.9981],
         [0.3059, 0.0967, 0.4872, 0.8695, 0.2324],
         [0.6734, 0.6645, 0.8859, 0.0637, 0.7380],
         [0.3292, 0.3411, 0.0896, 0.2175, 0.4013]],

        [[0.2818, 0.3043, 0.1920, 0.2259, 0.5281],
         [0.2549, 0.5776, 0.6986, 0.3627, 0.7794],
         [0.7877, 0.7343, 0.9829, 0.8692, 0.5039],
         [0.7503, 0.5880, 0.6443, 0.8743, 0.0187],
         [0.9316, 0.5449, 0.4706, 0.1604, 0.7445]]])
```

> Note: The result may not be exactly the same as the one displayed above, due to random number generation.

Though one and two-dimensional tensors are well-known, there are instances in machine learning where a three-dimensional tensor is essential.
````